util/src/org/commcare/util/CommCareTransactionParserFactory.java

```diff
@@ -0,0 +1,188 @@
+                    created = new FixtureXmlParser(parser) {
+                        //TODO: store these on the file system instead of in DB?
+                        private IStorageUtilityIndexed fixtureStorage;
+
+                        /*
+                         * (non-Javadoc)
+                         * @see org.commcare.xml.FixtureXmlParser#storage()
+                         */
+                        @Override
+                        public IStorageUtilityIndexed storage() {
+                            if (fixtureStorage == null) {
+                                fixtureStorage = CommCareTransactionParserFactory.this.sandbox.getUserFixtureStorage();
+                            }
+                            return fixtureStorage;
+                        }
+                    };
+                }
+
+                return created;
+            }
+        };
+
+        this.initUserParser();
+        this.initCaseParser();
+        this.initStockParser();
+    }
+
+
+    /* (non-Javadoc)
+     * @see org.commcare.data.xml.TransactionParserFactory#getParser(java.lang.String, java.lang.String, org.kxml2.io.KXmlParser)
+     */
+    public TransactionParser getParser(KXmlParser parser) {
+        String namespace = parser.getNamespace();
+        String name = parser.getName();
+        if (LedgerXmlParsers.STOCK_XML_NAMESPACE.matches(namespace)) {
+            if (stockParser == null) {
+                throw new RuntimeException("Couldn't process Stock transaction without initialization!");
+            }
+            req();
+            return stockParser.getParser(parser);
+        } else if (name != null && name.toLowerCase().equals("case")) {
+            if (caseParser == null) {
+                throw new RuntimeException("Couldn't receive Case transaction without initialization!");
+            }
+            req();
+            return caseParser.getParser(parser);
+        } else if (name != null && name.toLowerCase().equals("registration")) {
+            if (userParser == null) {
+                throw new RuntimeException("Couldn't receive User transaction without initialization!");
+            }
+            req();
+            return userParser.getParser(parser);
+        } else if (name != null && name.toLowerCase().equals("fixture")) {
+            req();
+            return fixtureParser.getParser(parser);
+        } else if (name != null && name.toLowerCase().equals("message")) {
+            //server message;
+            //" <message nature=""/>"
+        } else if (name != null && name.toLowerCase().equals("sync") && namespace != null && "http://commcarehq.org/sync".equals(namespace)) {
+            return new TransactionParser<String>(parser) {
+                /*
+                 * (non-Javadoc)
+                 * @see org.commcare.data.xml.TransactionParser#commit(java.lang.Object)
+                 */
+                @Override
+                public void commit(String parsed) throws IOException {
```

Comment:
I believe those are there so automated javadoccer's can redirect to the parent page, but we should verify.